pkg/plugins/golang/declarative/v1/api.go

```diff
@@ -97,27 +92,11 @@ func (p *createAPISubcommand) InjectResource(res *resource.Resource) error {
 func (p *createAPISubcommand) Scaffold(fs machinery.Filesystem) error {
 	fmt.Println("updating scaffold with declarative pattern...")
 
-	// Load the boilerplate
-	bp, err := afero.ReadFile(fs.FS, filepath.Join("hack", "boilerplate.go.txt"))
+	scaffolder := scaffolds.NewAPIScaffolder(p.config, *p.resource)
+	scaffolder.InjectFS(fs)
+	err := scaffolder.Scaffold()
 	if err != nil {
-		return fmt.Errorf("error updating scaffold: unable to load boilerplate: %w", err)
-	}
-	boilerplate := string(bp)
-
-	// Initialize the machinery.Scaffold that will write the files to disk
-	scaffold := machinery.NewScaffold(fs,
-		machinery.WithConfig(p.config),
-		machinery.WithBoilerplate(boilerplate),
-		machinery.WithResource(p.resource),
-	)
-
-	if err := scaffold.Execute(
-		&templates.Types{},
-		&templates.Controller{},
-		&templates.Channel{ManifestVersion: exampleManifestVersion},
-		&templates.Manifest{ManifestVersion: exampleManifestVersion},
-	); err != nil {
```

Comment:
This change shows that will broke the plugin and make it stop the controller and types scaffolds. 
Could you please see: https://github.com/kubernetes-sigs/kubebuilder/pull/2507/files#r800561564

Reply:
c/c @viveksyngh

Reply:
solved,